flink-runtime/src/main/java/org/apache/flink/runtime/deployment/TaskDeploymentDescriptorFactory.java

```diff
@@ -221,14 +224,17 @@ private ShuffleDescriptorAndIndex[] computeConsumedPartitionShuffleDescriptors(
     }
 
     private MaybeOffloaded<ShuffleDescriptorAndIndex[]> serializeAndTryOffloadShuffleDescriptor(
-            ShuffleDescriptorAndIndex[] shuffleDescriptors) throws IOException {
+            ShuffleDescriptorAndIndex[] shuffleDescriptors, int numConsumer) throws IOException {
```

Comment:
Can we abstract the logic here to a inline class `ShuffleDescriptorSerializer`, and provide 2 method: `serializeShuffleDescriptor` and `serializeAndTryOffloadShuffleDescriptor`. And then pass the serializer to `CachedShuffleDescriptors`, the `CachedShuffleDescriptors` will decide to call which method according to the `shouldOffload`